contributing/topics/schema-design-considerations.md

```diff
@@ -257,3 +257,159 @@ Would be better exposed as the following resources:
 - `azurerm_data_factory_linked_service_azure_function`
 - `azurerm_data_factory_linked_service_azure_search`
 
+## Preview Fields
+
+Fields that are in preview should not be supported until they reach General Availability (GA) status, as they may change or be removed before becoming stable.
+
+## Flattening nested properties
+
+When designing schemas, consider flattening properties with `MaxItems: 1` that contain only a single nested property unless the service team has confirmed additional nested properties are imminent. In those cases, add an inline comment explaining why the block is left unflattened so reviewers understand the rationale.
+
+:white_check_mark: **DO**
+```go
+"certificate": {
```

Comment:
should this not be
```suggestion
"credential_certificate": {
```
as we will often include the removed property in the new name? there could be multiple nested certificate properties